### PR TITLE
Add documentation on breadcrumbs

### PR DESCRIPTION
**Pull request by planger**
Contributed on behalf of STMicroelectronics.

### CODE REVIEW

**Review by JonasHelming — APPROVED** [review]

**Comment by JonasHelming** [comment]
@vince-fugnitto are you fine merging this?

File: src/docs/breadcrumbs.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+export class CustomWidget extends ReactWidget implements NavigatableWidget {
+    // ...
+    getResourceUri(): URI | undefined {
+        return new URI('resource://project-a/component-b/config-c');
+    }
+}
+```
+
+When this `CustomWidget` is opened, Theia will look for breadcrumbs contributions that return breadcrumbs for this URI.
+By default, it won't find any and thus won't show any breadcrums.
```
Comment (vince-fugnitto) [inline src/docs/breadcrumbs.md:36]: typo:

```suggestion
By default, it won't find any and thus won't show any breadcrumbs.
```

File: src/docs/breadcrumbs.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Once the breadcrumbs contribution is registered, Theia will show the following for our custom widget.
+
+<img src="/breadcrumbs-custom.png" alt="A screenshot of custom breadcrumbs in Theia" style="max-width: 525px">
+
+## Interactive Breadcrumbs
+
+Breadcrumbs contributions also must implement the method `attachPopupContent`.
+Theia invokes this method as soon as a user clicks on one of the breadcrumbs and passes a parent HTML element into which the breadcrumbs contribution can now add arbitrary HTML content, including trees, buttons, etc.
+This content is consequently being shown as a popup below the clicked breadcumbs element.
```
Comment (vince-fugnitto) [inline src/docs/breadcrumbs.md:98]: typo:

```suggestion
This content is consequently being shown as a popup below the clicked breadcrumbs element.
```

**Review by vince-fugnitto — APPROVED** [review]
LGTM :+1:

**Pull request merged**